### PR TITLE
Update runtime coop load / store, update rocwmma_coop API

### PR DESCRIPTION
**Pull request by cgmillette**
- Implements static_unroll to runtime arg versions of coop_load and coop_store
- Updates cooperative API: deprecate StrideCount arguments, expose template arg WaveCount functions
- Updates to API documentation

### CODE REVIEW

**Comment by cgmillette** [comment]
@CongMa13 @dlangbe @mkarunan can we review please?

File: library/include/rocwmma/rocwmma_coop.hpp [hunk 1]
```diff
@@ -84,7 +84,8 @@ namespace rocwmma
               uint32_t BlockK,
               typename DataT,
               typename DataLayout>
-    ROCWMMA_DEVICE void
+    [[deprecated("splitCount argument is deprecated and will be removed in a future "
```
Comment (CongMa13) [inline library/include/rocwmma/rocwmma_coop.hpp:87]: 👍

File: library/include/rocwmma/internal/coop_load.hpp [hunk 2]
```diff
@@ -158,7 +108,59 @@ namespace rocwmma
...
+        {
+            // Full fragment work
+            constexpr auto strideSpace = MatrixLayout::strideCounts();
+            constexpr auto strides     = MatrixLayout::strides();
+
+            // Drop the VW strides for splitting (reduced stride space).
+            constexpr auto strideSpaceR = pop_right(strideSpace);
+            constexpr auto stridesR     = pop_right(strides);
+            constexpr auto totalWorkItems
+                = flatten_coord_left((strideSpaceR - 1u), strideSpaceR) + 1u;
```
Comment (CongMa13) [inline library/include/rocwmma/internal/coop_load.hpp:125]: It took some time to understand how this expression works. We can add a new tuple multiply function later. Like the `add` lambda in this function.  :D

```
constexpr auto mul = [](auto... items) { return (items * ...); };
auto   totalWorkItems = std::apply(mul, strideSpaceR);
```

File: library/include/rocwmma/rocwmma_coop.hpp [hunk 3]
```diff
@@ -160,10 +159,88 @@ namespace rocwmma
...
+    //!
+    //! The full load is split into work items (SplitCount).
+    //! Work items are assigned in round robin fashion to waves in the range of [0, waveCount).
+    //! The current wave index determines the order of the current wave in the collaboration pool.
+    //! Work items are consumed in order by waves [0, waveCount) until
+    //! there are no more work items and the operation is completed.
+    /*!
+      \param frag Fragment of type MatrixT with its associated block sizes, data type and layout
+      \param data Data pointer to global/local memory
+      \param ldm Leading dimension size
```
Comment (CongMa13) [inline library/include/rocwmma/rocwmma_coop.hpp:179]: seems like waveIndex is missed here.

**Review by CongMa13 — APPROVED** [review]
I love the new coop. The design can be easily seen in the code.

**Review by mkarunan — APPROVED** [review]

File: library/include/rocwmma/rocwmma_coop.hpp (hunk 3 above)
Comment (cgmillette) [reply library/include/rocwmma/rocwmma_coop.hpp:179]: Done! Thanks

File: library/include/rocwmma/internal/coop_load.hpp (hunk 2 above)
Comment (cgmillette) [reply library/include/rocwmma/internal/coop_load.hpp:125]: On the TODO list!

**Pull request merged**